src/reports/components/new-tab-link-confirmation-dialog.tsx

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { NewTabLink } from 'common/components/new-tab-link';
+import { NamedFC } from 'common/react/named-fc';
+import { getId, ILinkProps } from 'office-ui-fabric-react';
+import * as React from 'react';
+
+export const NewTabLinkWithConfirmationDialog = NamedFC<ILinkProps>(
+    'NewTabLinkWithConfirmationDialog',
+    props => {
+        const id = getId('new-tab-link-with-confirmation-dialog__'); // generates something like new-tab-link-with-confirmation-dialog__123
+
+        // tslint:disable: react-no-dangerous-html
```

Comment:
I think this needs to be:
```typescript
// tslint:disable-next-line:react-no-dangerous-html
```
(notice the `next-line` in the comment)

Also (but not for this PR), I also had to disable this same rule on other places around the report. Should we think on a better strategy so we can avoid grandfathering new usages of this prop?

Reply:
Would you be okay with the comment looking like this:
```
{/* tslint:disable-next-line: react-no-dangerous-html */}
```

I saw `tslint:disable` without `-next-line` in `results-container.tsx` too

I agree that we should try to come up with a better strategy for this in a separate PR

Reply:
So, using `// tslint:disable:react-no-dangerous-html` (without `-next-line`) disables the rule for the rest of the file. I think in general we should avoid using that on our code base. The usage you found on `results-container.tsx` should be fixed.
Using `// tslint:disable-next-line:react-no-dangeours-html` does what you should expect: disable that rule only on the next line thus forcing us to be explicit when we want to do something "against the rules" (this may come handy if at any time we'd like to have a little script to count all the instances of this exceptions we have).

As per the comment syntax, I see what you mean and that's fine. The comment style should not matter as long as we can tell tslint to ignore that rule on that specific line.

Reply:
Changed for both files